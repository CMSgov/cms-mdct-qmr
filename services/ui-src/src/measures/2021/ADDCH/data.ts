import { DataDrivenTypes } from "measures/2021/CommonQuestions/types";
import * as DC from "dataConstants";

/**
 * Attention
 * Changing the keys of these objects will change how the measure data is shaped and should not be done unless that is the desired result.
 * Changing the values of these objects will change the text that is displayed to the user.
 */
export const qualifierLabelsAndText = {
  "Initiation Phase": "Initiation Phase",
  "Continuation and Maintenance (C&M) Phase":
    "Continuation and Maintenance (C&M) Phase",
};
export const categoryLabelsAndText = {};

export const qualifiers = Object.keys(qualifierLabelsAndText);
export const categories = Object.keys(categoryLabelsAndText);

export const data: DataDrivenTypes.PerformanceMeasure = {
  questionText: [
    "Percentage of children newly prescribed attention-deficit/hyperactivity disorder (ADHD) medication who had at least three follow-up care visits within a 10-month period, one of which was within 30 days of when the first ADHD medication was dispensed. Two rates are reported.",
  ],
  questionSubtextTitles: [
    "Initiation Phase",
    "Continuation and Maintenance (C&M) Phase",
  ],
  questionSubtext: [
    "Percentage of children ages 6 to 12 with an ambulatory prescription dispensed for ADHD medication, who had one follow-up visit with a practitioner with prescribing authority during the 30-day Initiation Phase.",
    "Percentage of children ages 6 to 12 with an ambulatory prescription dispensed for ADHD medication who remained on the medication for at least 210 days and who, in addition to the visit in the Initiation Phase, had at least two follow-up visits with a practitioner within 270 days (9 months) after the Initiation Phase ended.",
  ],
  categories,
  qualifiers,
};

export const dataSourceData: DataDrivenTypes.DataSource = {
  optionsLabel:
    "If reporting entities (e.g., health plans) used different data sources, please select all applicable data sources used below.",
  options: [
    {
      value: DC.ADMINISTRATIVE_DATA,
      subOptions: [
        {
          label: "What is the Administrative Data Source?",
          options: [
            {
              value: DC.MEDICAID_MANAGEMENT_INFO_SYSTEM,
            },

            {
              value: DC.ADMINISTRATIVE_DATA_OTHER,
              description: true,
            },
          ],
        },
      ],
    },
    {
      value: DC.ELECTRONIC_HEALTH_RECORDS,
      description: true,
    },
    {
      value: DC.OTHER_DATA_SOURCE,
      description: true,
    },
  ],
};
