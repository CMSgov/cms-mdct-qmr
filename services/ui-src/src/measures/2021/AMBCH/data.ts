import { DataDrivenTypes } from "measures/2021/CommonQuestions/types";

export const qualifiers = [
  "< Age 1",
  "Ages 1 to 9",
  "Ages 10 to 19",
  "Ages unknown",
  "Total (Ages <1 to 19)",
];
export const categories = [];

export const data: DataDrivenTypes.PerformanceMeasure = {
  questionText: [
    "Rate of emergency department (ED) visits per 1,000 beneficiary months among children up to age 19.",
  ],
  categories,
  qualifiers,
};
