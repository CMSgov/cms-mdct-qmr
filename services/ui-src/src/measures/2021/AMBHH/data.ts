import { DataDrivenTypes } from "measures/CommonQuestions/types";

export const qualifiers = [
  "Ages 0 to 17",
  "Ages 18 to 64",
  "Age 65 and older",
  "Ages unknown",
  "Total",
];
export const categories = [];

export const data: DataDrivenTypes.PerformanceMeasure = {
  questionText: [
    "Rate of emergency department (ED) visits per 1,000 enrollee months among Health Home enrollees.",
  ],
  categories,
  qualifiers,
};
