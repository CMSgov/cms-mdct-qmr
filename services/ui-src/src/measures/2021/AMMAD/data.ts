import { DataDrivenTypes } from "measures/2021/CommonQuestions/types";

/**
 * Attention
 * Changing the keys of these objects will change how the measure data is shaped and should not be done unless that is the desired result.
 * Changing the values of these objects will change the text that is displayed to the user.
 */
export const qualifierLabelsAndText = {
  "Ages 18 to 64": "Ages 18 to 64 - here is the text i would like displayed",
  "Age 65 and older": "Age 65 and older",
};
export const categoryLabelsAndText = {
  "Effective Acute Phase Treatment":
    "Effective Acute Phase Treatment - here is some custom text for a category",
  "Effective Continuation Phase Treatment":
    "Effective Continuation Phase Treatment",
};

export const qualifiers = Object.keys(qualifierLabelsAndText);
export const categories = Object.keys(categoryLabelsAndText);

export const data: DataDrivenTypes.PerformanceMeasure = {
  questionText: [
    "Percentage of beneficiaries age 18 and older who were treated with antidepressant medication, had a diagnosis of major depression, and who remained on an antidepressant medication treatment. Two rates are reported:",
  ],
  questionListItems: [
    "Effective Acute Phase Treatment: Percentage of beneficiaries who remained on an antidepressant medication for at least 84 days (12 weeks).",
    "Effective Continuation Phase Treatment: Percentage of beneficiaries who remained on an antidepressant medication for at least 180 days (6 months).",
  ],
  categories,
  qualifiers,
};

export const dataSourceData: DataDrivenTypes.DataSource = {
  optionsLabel:
    "If reporting entities (e.g., health plans) used different data sources, please select all applicable data sources used below.",
  options: [
    {
      value: "Administrative Data",
      subOptions: [
        {
          label: "What is the Administrative Data Source?",
          options: [
            {
              value: "Medicaid Management Information System (MMIS)",
            },
            {
              value: "Administrative Data Other",
              description: true,
            },
          ],
        },
      ],
    },
    {
      value: "Electronic Health Records",
      description: true,
    },
    {
      value: "Other Data Source",
      description: true,
    },
  ],
};
