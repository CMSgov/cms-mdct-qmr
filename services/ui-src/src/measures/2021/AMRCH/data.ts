import { DataDrivenTypes } from "measures/2021/CommonQuestions/types";
import { getRateInfo } from "utils";

export const { categories, qualifiers } = getRateInfo();

export const data: DataDrivenTypes.PerformanceMeasure = {
  questionText: [
    "The percentage of children and adolescents ages 5 to 18 who were identified as having persistent asthma and had a ratio of controller medications to total asthma medications of 0.50 or greater during the measurement year.",
  ],
  categories,
  qualifiers,
};
