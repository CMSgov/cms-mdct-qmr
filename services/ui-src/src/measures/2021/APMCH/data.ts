import { DataDrivenTypes } from "measures/2021/CommonQuestions/types";
import { getRateInfo } from "utils";

export const { categories, qualifiers } = getRateInfo();

export const data: DataDrivenTypes.PerformanceMeasure = {
  questionText: [
    "Percentage of children and adolescents ages 1 to 17 who had two or more antipsychotic prescriptions and had metabolic testing. Three rates are reported:",
  ],
  questionListItems: [
    "Percentage of children and adolescents on antipsychotics who received blood glucose testing",
    "Percentage of children and adolescents on antipsychotics who received cholesterol testing",
    "Percentage of children and adolescents on antipsychotics who received blood glucose and cholesterol testing",
  ],
  categories,
  qualifiers,
};
