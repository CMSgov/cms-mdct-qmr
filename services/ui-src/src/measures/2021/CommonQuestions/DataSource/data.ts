export interface OptionNode {
  value: string;
  subOptions?: {
    label?: string;
    options: OptionNode[];
  };
}

export interface DataSourceData {
  options: OptionNode[];
  optionsLabel: string;
}

export const defaultData: DataSourceData = {
  optionsLabel:
    "If reporting entities (e.g., health plans) used different data sources, please select all applicable data sources used below.",
  options: [
    {
      value: "Administrative Data",
      subOptions: {
        label: "What is the Administrative Data Source?",
        options: [
          {
            value: "Medicaid Management Information System (MMIS)",
          },
          {
            value: "Administrative Data Other",
          },
        ],
      },
    },
    {
      value: "Other Data Source",
    },
  ],
};
