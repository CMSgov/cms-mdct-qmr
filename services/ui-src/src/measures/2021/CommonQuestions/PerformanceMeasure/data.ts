export interface PerformanceMeasureData {
  qualifiers?: string[]; // age ranges, etc
  categories?: string[]; //performance measure descriptions
  questionText?: string;
  questionListItems?: string[];
}

/** Example data built from IETAD */
export const exampleData: PerformanceMeasureData = {
  questionText:
    "The percentage of beneficiaries age 18 and older with a new episode of alcohol or other drug (AOD) abuse or dependence who received the following",
  questionListItems: [
    "Initiation of AOD Treatment: Percentage of beneficiaries who initiate treatment through an inpatient AOD admission, outpatient visit, intensive outpatient encounter, or partial hospitalization, telehealth, or medication assisted treatment within 14 days of the diagnosis.",
    "Engagement of AOD Treatment: Percentage of beneficiaries who initiated treatment and who were engaged in ongoing AOD treatment within 34 days of the initiation visit.",
  ],
  qualifiers: ["Ages 18 to 64", "Age 65 and older"],
  categories: [
    "Initiation of AOD Treatment: Alcohol Abuse or Dependence",
    "Engagement of AOD Treatment: Alcohol Abuse or Dependence",
    "Initiation of AOD Treatment: Opioid Abuse or Dependence",
    "Engagement of AOD Treatment: Opioid Abuse or Dependence",
    "Initiation of AOD Treatment: Other Drug Abuse or Dependence",
    "Engagement of AOD Treatment: Other Drug Abuse or Dependence",
    "Initiation of AOD Treatment: Total AOD Abuse or Dependence",
    "Engagement of AOD Treatment: Total AOD Abuse or Dependence",
  ],
};
