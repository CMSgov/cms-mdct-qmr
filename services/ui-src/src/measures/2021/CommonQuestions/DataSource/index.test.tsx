import * as DC from "dataConstants";
import { DataSource } from "./index";
import { DataSourceData } from "./data";
import fireEvent from "@testing-library/user-event";
import { renderWithHookForm } from "utils/testUtils/reactHookFormRenderer";
import { screen } from "@testing-library/react";

describe("Test the global DataSource component", () => {
  describe("with default data options", () => {
    const DataSourceComponent = <DataSource />;
    const renderComponent = () => renderWithHookForm(DataSourceComponent);

    it("Check that the component renders correctly", async () => {
      const { container } = renderComponent();

      expect(screen.getByText("Data Source")).toBeVisible();

      // there should be only 2 options
      expect(container.getElementsByClassName("chakra-checkbox").length).toBe(
        2
      );

      const option1 = screen.getByLabelText(DC.ADMINISTRATIVE_DATA);
      const option2 = screen.getByLabelText(DC.OTHER_DATA_SOURCE);

      // Clicking option1 should reveal 2 sub-options
      fireEvent.click(option1);
      const option1Children = container.querySelectorAll(
        ".chakra-stack .css-n21gh5 > *"
      );
      expect(option1Children.length).toBe(2);

      expect(
        screen.getByText(DC.MEDICAID_MANAGEMENT_INFO_SYSTEM)
      ).toBeInTheDocument();
      expect(
        screen.getByText(DC.ADMINISTRATIVE_DATA_OTHER)
      ).toBeInTheDocument();

      // should do nothing
      fireEvent.click(
        screen.getByLabelText(DC.MEDICAID_MANAGEMENT_INFO_SYSTEM)
      );

      expect(
        screen.queryAllByLabelText("Describe the data source:").length
      ).toBe(0);

      // should reveal text area
      fireEvent.click(screen.getByLabelText(DC.ADMINISTRATIVE_DATA_OTHER));

      const adminDataOtherTextArea = await screen.findByLabelText(
        "Describe the data source:"
      );
      fireEvent.type(adminDataOtherTextArea, "This is the test text");
      expect(adminDataOtherTextArea).toHaveDisplayValue(
        "This is the test text"
      );

      // close option1
      fireEvent.click(option1);

      // Clicking option2 should reveal a text area
      fireEvent.click(option2);
      const textArea = await screen.findByLabelText(
        "Describe the data source:"
      );
      fireEvent.type(textArea, "This is the test text");
      expect(textArea).toHaveDisplayValue("This is the test text");
    });
  });

  describe("with custom data options", () => {
    // define custom data options
    const data: DataSourceData = {
      optionsLabel: "",
      options: [
        {
          value: "Option 1",
          subOptions: [
            {
              label: "What is the Option 1 Data Source?",
              options: [
                {
                  value: "I'm not telling",
                },
                {
                  value: "You'll have to kill me first",
                  description: true,
                },
              ],
            },
          ],
        },
        {
          value: "A Better Option",
          description: true,
        },
        {
          value: "Something Even Crazier Than The First 2",
          description: false,
        },
      ],
    };
    const DataSourceComponent = <DataSource data={data} />;
    const renderComponent = () => renderWithHookForm(DataSourceComponent);

    it("Check that the component renders correctly", async () => {
      const { container } = renderComponent();

      expect(screen.getByText("Data Source")).toBeVisible();

      // check that there should be only 2 options
      expect(container.getElementsByClassName("chakra-checkbox").length).toBe(
        3
      );

      const option1 = screen.getByLabelText("Option 1");
      const option2 = screen.getByLabelText("A Better Option");
      const option3 = screen.getByLabelText(
        "Something Even Crazier Than The First 2"
      );

      // Clicking option1 should reveal 2 sub-options
      fireEvent.click(option1);
      const option1Children = container.querySelectorAll(
        ".chakra-stack .css-n21gh5 > *"
      );
      expect(option1Children.length).toBe(2);

      expect(screen.getByText("I'm not telling")).toBeInTheDocument();
      expect(
        screen.getByText("You'll have to kill me first")
      ).toBeInTheDocument();

      // should do nothing
      fireEvent.click(screen.getByLabelText("I'm not telling"));

      expect(
        screen.queryAllByLabelText("Describe the data source:").length
      ).toBe(0);

      // should reveal text area
      fireEvent.click(screen.getByLabelText("You'll have to kill me first"));

      const subOptionTextArea = await screen.findByLabelText(
        "Describe the data source:"
      );
      fireEvent.type(subOptionTextArea, "This is the test text");
      expect(subOptionTextArea).toHaveDisplayValue("This is the test text");

      // close option1
      fireEvent.click(option1);

      // Clicking option2 should reveal a text area
      fireEvent.click(option2);
      const option2TextArea = await screen.findByLabelText(
        "Describe the data source:"
      );
      fireEvent.type(option2TextArea, "This is the test text");
      expect(option2TextArea).toHaveDisplayValue("This is the test text");

      // close option2
      fireEvent.click(option2);

      // Clicking option3 do nothing
      fireEvent.click(option3);
      expect(
        screen.queryAllByLabelText("Describe the data source:").length
      ).toBe(0);

      fireEvent.click(option3);
    });
  });
});
