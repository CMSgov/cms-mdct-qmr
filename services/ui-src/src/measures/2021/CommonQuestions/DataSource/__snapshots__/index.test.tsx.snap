// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the global DataSource component (Custom Structure) Component renders with correct content 1`] = `
<body>
  <div
    class="css-pb69ky"
  >
    <label
      class="chakra-form__label css-e3y4kl"
      data-cy="Data Source"
    >
      Data Source
    </label>
    <div
      data-cy="data-source-options"
    >
      <div
        class="chakra-form-control css-0"
        role="group"
      >
        <label
          class="chakra-form__label css-uvho5g"
          data-cy="How do we feel about a label with this text? Does it work?"
          for="field-2"
          id="field-2-label"
        >
          How do we feel about a label with this text? Does it work?
        </label>
        <div
          class="chakra-stack css-n21gh5"
        >
          <div
            class="css-79elbk"
          >
            <label
              class="chakra-checkbox css-1uiwwan"
              data-cy="DataSource0"
            >
              <input
                class="chakra-checkbox__input"
                id="DataSource0-checkbox"
                style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                type="checkbox"
                value="Option1"
              />
              <span
                aria-hidden="true"
                class="chakra-checkbox__control css-dnty2r"
              />
              <span
                class="chakra-checkbox__label css-1oeb2oe"
              >
                <p
                  class="chakra-text css-vjer7o"
                >
                  Option 1
                </p>
              </span>
            </label>
            <div
              class="chakra-collapse"
              style="overflow: hidden; display: none; opacity: 0; height: 0px;"
            />
          </div>
          <div
            class="css-79elbk"
          >
            <label
              class="chakra-checkbox css-1uiwwan"
              data-cy="DataSource1"
            >
              <input
                class="chakra-checkbox__input"
                id="DataSource1-checkbox"
                style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                type="checkbox"
                value="ABetterOption"
              />
              <span
                aria-hidden="true"
                class="chakra-checkbox__control css-dnty2r"
              />
              <span
                class="chakra-checkbox__label css-1oeb2oe"
              >
                <p
                  class="chakra-text css-vjer7o"
                >
                  A Better Option
                </p>
              </span>
            </label>
            <div
              class="chakra-collapse"
              style="overflow: hidden; display: none; opacity: 0; height: 0px;"
            />
          </div>
          <div
            class="css-79elbk"
          >
            <label
              class="chakra-checkbox css-1uiwwan"
              data-cy="DataSource2"
            >
              <input
                class="chakra-checkbox__input"
                id="DataSource2-checkbox"
                style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                type="checkbox"
                value="SomethingEvenCrazierThanTheFirst2"
              />
              <span
                aria-hidden="true"
                class="chakra-checkbox__control css-dnty2r"
              />
              <span
                class="chakra-checkbox__label css-1oeb2oe"
              >
                <p
                  class="chakra-text css-vjer7o"
                >
                  Something Even Crazier Than The First 2
                </p>
              </span>
            </label>
            <div
              class="chakra-collapse"
              style="overflow: hidden; display: none; opacity: 0; height: 0px;"
            />
          </div>
        </div>
        <div
          class="css-k008qs"
        >
          <div
            class="css-17xejub"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Test the global DataSource component (Default) Component renders with correct content 1`] = `
<body>
  <div
    class="css-pb69ky"
  >
    <label
      class="chakra-form__label css-e3y4kl"
      data-cy="Data Source"
    >
      Data Source
    </label>
    <div
      data-cy="data-source-options"
    >
      <div
        class="chakra-form-control css-0"
        role="group"
      >
        <label
          class="chakra-form__label css-uvho5g"
          data-cy="If reporting entities (e.g., health plans) used different data sources, please select all applicable data sources used below."
          for="field-1"
          id="field-1-label"
        >
          If reporting entities (e.g., health plans) used different data sources, please select all applicable data sources used below.
        </label>
        <div
          class="chakra-stack css-n21gh5"
        >
          <div
            class="css-79elbk"
          >
            <label
              class="chakra-checkbox css-1uiwwan"
              data-cy="DataSource0"
            >
              <input
                class="chakra-checkbox__input"
                id="DataSource0-checkbox"
                style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                type="checkbox"
                value="AdministrativeData"
              />
              <span
                aria-hidden="true"
                class="chakra-checkbox__control css-dnty2r"
              />
              <span
                class="chakra-checkbox__label css-1oeb2oe"
              >
                <p
                  class="chakra-text css-vjer7o"
                >
                  Administrative Data
                </p>
              </span>
            </label>
            <div
              class="chakra-collapse"
              style="overflow: hidden; display: none; opacity: 0; height: 0px;"
            />
          </div>
          <div
            class="css-79elbk"
          >
            <label
              class="chakra-checkbox css-1uiwwan"
              data-cy="DataSource1"
            >
              <input
                class="chakra-checkbox__input"
                id="DataSource1-checkbox"
                style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                type="checkbox"
                value="OtherDataSource"
              />
              <span
                aria-hidden="true"
                class="chakra-checkbox__control css-dnty2r"
              />
              <span
                class="chakra-checkbox__label css-1oeb2oe"
              >
                <p
                  class="chakra-text css-vjer7o"
                >
                  Other Data Source
                </p>
              </span>
            </label>
            <div
              class="chakra-collapse"
              style="overflow: hidden; display: none; opacity: 0; height: 0px;"
            />
          </div>
        </div>
        <div
          class="css-k008qs"
        >
          <div
            class="css-17xejub"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
